frontend/src/components/tracing/tracing.tsx

```diff
@@ -0,0 +1,391 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+import { FAVICONS } from "@/components/editor/dynamic-favicon";
+import React, { useRef, useState } from "react";
+import type { CellId } from "@/core/cells/ids";
+import { ElapsedTime, formatElapsedTime } from "../editor/cell/CellStatus";
+import { Tooltip } from "@/components/ui/tooltip";
+import { type Config, type TopLevelSpec, compile } from "vega-lite";
+import { ChevronRight, ChevronDown, SettingsIcon } from "lucide-react";
+import type { VisualizationSpec } from "react-vega";
+
+const LazyVegaLite = React.lazy(() =>
+  import("react-vega").then((m) => ({ default: m.VegaLite })),
+);
+
+interface CellRun {
+  cellID: string;
+  code: string;
+  elapsedTime: number;
+  status: RunStatus;
+}
+
+export interface Run {
```

Comment:
Do you want me to connect this to the backend? i get the backend to return a runId for each cell-op, and get you setup so you can pull this data from state

Reply:
yeah! hmm, I thought it would be more complex but that should work. I'm happy to try connecting it myself 🤔 unless you foresee difficult challenges or decisions.